maintenance/pin.sh

```diff
@@ -1,9 +1,8 @@
 #!/usr/bin/env bash
 # We run this in balrog/. to pin all requirements.
 #
-# Usage: maintenance/pin.sh [<extra-pip-compile-multi-arguments>]
+# Usage: maintenance/pin.sh [<extra-uv-arguments>]
 #
-set -e
-set -x
+set -ex
 
-docker run --platform=linux/x86_64 --rm -t -v $PWD:/src -w /src python:3.13 maintenance/pin-helper.sh $@
+uv lock -U $@
```

Comment:
Personally I'd prefer we delete this file and document / get more comfortable using uv directly.. but I guess there's not really any harm in leaving it.

Reply:
+1 - we should just have an entry in README.md, there's already a maintenance section there:
>To update the python dependencies, use ./maintenance/pin.sh, do not run pip-compile-multi manually.

Reply:
I did this to keep consistency but I'm more than fine with removing more stuff :D